File: MARCH 2024/09_03_2024.py
class Solution:
    def nthCharacter(self, s, r, n):
        
        m=len(s)
        ans=s
        itr=int(0)
        if(m%2==0):
            itr=int(m/2)
        else:
            itr=int((m/2)+1)
            
        for i in range(0,r):
            
            temp=""
            for j in range(0,itr):
                
                if(ans[j]=="1"):
                    temp+="10"
                    
                else:
                    temp+="01"
            
            ans=temp
            
        return ans[n]

File: MARCH 2024/09_03_2024.cpp
class Solution{
  public:
    char nthCharacter(string s, int r, int n) {
        
        int m=s.size();
        string ans=s;
        
        int itr=m%2==0?m/2:(m/2)+1;
        
        for(int i=0;i<r;i++){
            
            string temp="";
            for(int j=0;j<itr;j++){
                
                if(ans[j]=='1'){
                    
                    temp+="10";
                }
                
                else{
                    temp+="01";
                }
            }
            
            ans=temp;
        }
        
        return ans[n];
    }
};
